### PR TITLE
Problem: chain-id is not parsed from genesis config

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- simpliy use the default behavior in sdk

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### Summary by CodeRabbit

- Bug Fix: Resolved an issue with concurrent write in fee history, enhancing the reliability of transactions.
- Refactor: Simplified the chain ID setup process in the SDK, removing unnecessary code for a cleaner, more efficient system. This change streamlines the underlying processes, but won't affect the user experience directly.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## UNRELEASED
+
+### Bug Fixes
+
+- [#]() Use the default chain-id behavour in sdk.
```
Comment (yihuang) [inline CHANGELOG.md:7]:
```suggestion
- [#1217](https://github.com/crypto-org-chain/cronos/pull/1217) Use the default chain-id behavour in sdk.
```

File: cmd/cronosd/cmd/root.go [hunk 2]
```diff
@@ -243,36 +241,14 @@ type appCreator struct {
...
 	skipUpgradeHeights := make(map[int64]bool)
 	for _, h := range cast.ToIntSlice(appOpts.Get(server.FlagUnsafeSkipUpgrades)) {
 		skipUpgradeHeights[int64(h)] = true
 	}
 
-	// Setup chainId
-	chainID := cast.ToString(appOpts.Get(flags.FlagChainID))
-	if len(chainID) == 0 {
-		v := viper.New()
-		v.AddConfigPath(filepath.Join(home, "config"))
```
Comment (mmsqe) [inline cmd/cronosd/cmd/root.go:260]: seems we cant add it in appOpts, need fallback to gensis at least

**Review by mmsqe — APPROVED** [review]

File: cmd/cronosd/cmd/root.go (hunk 2 above)
Comment (yihuang) [reply cmd/cronosd/cmd/root.go:260]: config.toml is merged into appOpts I think

Comment (mmsqe) [reply cmd/cronosd/cmd/root.go:260]: we only config in client.toml not in config.toml

Comment (yihuang) [reply cmd/cronosd/cmd/root.go:260]: `client.toml` too

**Pull request merged**